source/common/router/router.cc

```diff
@@ -673,7 +673,10 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
       headers, callbacks_->streamInfo().downstreamAddressProvider().sslConnection() != nullptr);
 
   // Ensure an http transport scheme is selected before continuing with decoding.
-  ASSERT(headers.Scheme());
+  if (!headers.Scheme()) {
```

Comment:
Honestly I think we can leave this assert as-is or simply remove it it's basically validating that setUpstreamScheme sets a scheme, which it deterministically does.

Reply:
OK then I would probably just leave it and close this PR?

Reply:
@alyssawilk could you help check #14961, if all are like this, maybe we can close that as well.

Reply:
kindly ping @alyssawilk

Reply:
yeah I mean I'm generally a fan of asserts so I'd be inclined to close off that issue, sorry